tests/consumers/test_consumer_builder.py

```diff
@@ -0,0 +1,254 @@
+)
+
+# Below, a ConsumerBuilder with all required
+# and optional args, but only those with
+# no default values
+consumer_builder_with_opt = ConsumerBuilder(
+    storage_key=test_storage_key,
+    kafka_params=optional_kafka_params,
+    processing_params=ProcessingParameters(
+        processes=5,
+        input_block_size=6,
+        output_block_size=7,
+    ),
+    max_batch_size=3,
+    max_batch_time_ms=4,
+    metrics=MetricsWrapper(
+        environment.metrics,
+        "test_consumer",
+        tags={"group": consumer_group_name, "storage": test_storage_key.value},
+    ),
+    parallel_collect=False,
+    slice_id=None,
+)
+
+
+def test_consumer_builder_non_optional_attributes() -> None:
+    # Ensures that the ConsumerBuilders are assigning a
+    # not-None value to the required attributes
+
+    # Depending on the attribute, we can verify this
+    # to different degrees
+
+    assert consumer_builder.storage == get_writable_storage(test_storage_key)
+    assert consumer_builder_with_opt.storage == get_writable_storage(test_storage_key)
+
+    assert consumer_builder.consumer_group == consumer_group_name
+    assert consumer_builder_with_opt.consumer_group == consumer_group_name
+
+    assert isinstance(consumer_builder.raw_topic, Topic)
+    assert isinstance(consumer_builder_with_opt.raw_topic, Topic)
+
+    assert consumer_builder.broker_config is not None
+    assert consumer_builder_with_opt.broker_config is not None
+
+    assert consumer_builder.producer_broker_config is not None
+    assert consumer_builder_with_opt.producer_broker_config is not None
+
+    assert isinstance(consumer_builder.producer, Producer)
+    assert isinstance(consumer_builder_with_opt.producer, Producer)
+
+    assert isinstance(consumer_builder.metrics, MetricsBackend)
+    assert isinstance(consumer_builder_with_opt.metrics, MetricsBackend)
+
+    assert consumer_builder.max_batch_size == 3
+    assert consumer_builder.max_batch_time_ms == 4
+    assert consumer_builder.auto_offset_reset == "earliest"
+    assert consumer_builder.queued_max_messages_kbytes == 1
+    assert consumer_builder.queued_min_messages == 2
+
+    assert consumer_builder_with_opt.max_batch_size == 3
+    assert consumer_builder_with_opt.max_batch_time_ms == 4
+    assert consumer_builder_with_opt.auto_offset_reset == "earliest"
+    assert consumer_builder_with_opt.queued_max_messages_kbytes == 1
+    assert consumer_builder_with_opt.queued_min_messages == 2
+
+
+def test_consumer_builder_optional_attributes() -> None:
+    # Ensures that the ConsumerBuilders are assigning
+    # some value, None or not, to the optional attributes
+
+    # In the case that optional Kafka topic overrides
+    # are passed in, stronger checks are performed
+    # in a separate test
+
+    try:
+        consumer_builder.bootstrap_servers
+    except NameError:
+        assert (
+            False
+        ), "ConsumerBuilder's bootstrap_servers must either be defined as some value or None"
```

Comment:
How about removing the try/except/assert False. Makes it easier to keep these up to date if the attributes change.
```suggestion
    consumer_builder.bootstrap_servers
```

Reply:
This works!